nengo_loihi/simulator.py

```diff
@@ -387,11 +396,16 @@ def __init__(  # noqa: C901
             discretize_model(self.model)
 
         if target in ("simreal", "sim"):
+            if hw_opts is not None:
+                raise ValidationError("hw_opts should be None unless "
```

Comment:
I would either warn or do nothing here. While it could sometimes be indicative of a misunderstanding if the user passes hardware args when `target='sim'`, it could also just be that they like to switch back and forth, and it would be nice if they didn't have to worry about getting rid of the hardware args in this case. I would probably vote for warn, though I don't have strong feelings either way.

Reply:
I guess one argument for having it as an error is as a safe-guard to prevent people from accidentally writing unit tests that use different allocators with the emulator, which would be a waste of CPU time. Warnings can easily slip through passing unit tests undetected. Otherwise I don't think it's immediately obvious that the emulator isn't doing something to emulate the allocation in the event of there being some practical difference.

Reply:
There's a lot of bad tests that can be written. I'm not convinced we need to hand-hold devs in this way. Also, we try to check the test warnings somewhat regularly and reduce obvious ones.

I just know I often switch back and forth between emulator and hardware. And for tests, then we can use the `Simulator` fixture if e.g. we want to run on emulator and hardware, but on hardware we need `snip_max_spikes_per_step=200` or something. Actually, now that I say this, I'm starting to convince myself that we shouldn't have a warning, either.

Reply:
Resolved by removing error/warning.